lib/services/project-changes-service.ts

```diff
@@ -218,20 +219,20 @@ export class ProjectChangesService implements IProjectChangesService {
 	}
 
 	private containsNewerFiles(dir: string, skipDir: string, projectData: IProjectData, processFunc?: (filePath: string, projectData: IProjectData) => boolean): boolean {
+		const dirFileStat = this.$fs.getFsStats(dir);
+		if(this.isFileModified(dirFileStat, dir)) {
+			return true;
+		}
+
 		let files = this.$fs.readDirectory(dir);
 		for (let file of files) {
 			let filePath = path.join(dir, file);
 			if (filePath === skipDir) {
 				continue;
 			}
 
-			let fileStats = this.$fs.getFsStats(filePath);
-
-			let changed = fileStats.mtime.getTime() >= this._outputProjectMtime || fileStats.ctime.getTime() >= this._outputProjectCTime;
-			if (!changed) {
-				let lFileStats = this.$fs.getLsStats(filePath);
-				changed = lFileStats.mtime.getTime() >= this._outputProjectMtime || lFileStats.ctime.getTime() >= this._outputProjectCTime;
-			}
+			const fileStats = this.$fs.getFsStats(filePath);
```

Comment:
It's used [here](https://github.com/NativeScript/nativescript-cli/pull/3042/files/9020a8648f41416cec30c7c45456033c33b96db7#diff-919912d52b0e0303e76febe7c84cacfdR249)